ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -341,59 +342,53 @@ runM Interfaces
+                  timeWaitTimeout     = local_TIME_WAIT_TIMEOUT,
+                  outboundIdleTimeout = local_PROTOCOL_IDLE_TIMEOUT,
+                  connectionDataFlow  = ntcDataFlow,
+                  prunePolicy         = Diffusion.Policies.prunePolicy,
+                  stdGen              = cmLocalStdGen,
+                  connectionsLimits   = localConnectionLimits,
+                  updateVersionData   = \a _ -> a,
+                  connStateIdSupply   = diConnStateIdSupply,
+                  classifyHandleError
               }
-
-        CM.with
-          localConnectionManagerArguments
-          localConnectionHandler
-          classifyHandleError
-          (InResponderMode localInbInfoChannel)
-          $ \localConnectionManager-> do
-            --
-            -- run node-to-client server
-            --
-            traceWith tracer . RunLocalServer
-              =<< Snocket.getLocalAddr diNtcSnocket localSocket
-
-            Server.with
-              Server.Arguments {
-                  Server.sockets               = localSocket :| [],
-                  Server.snocket               = diNtcSnocket,
-                  Server.tracer                = dtLocalServerTracer,
-                  Server.trTracer              = nullTracer, -- TODO: issue #3320
-                  Server.debugInboundGovernor  = nullTracer,
-                  Server.inboundGovernorTracer = dtLocalInboundGovernorTracer,
-                  Server.inboundIdleTimeout    = Nothing,
-                  Server.connectionLimits      = localConnectionLimits,
-                  Server.connectionManager     = localConnectionManager,
-                  Server.connectionDataFlow    = ntcDataFlow,
-                  Server.inboundInfoChannel    = localInbInfoChannel
-                }
-              (\inboundGovernorThread _ -> Async.wait inboundGovernorThread)
+              (InResponderMode responderInfoChannel)
+              connectionHandler
+              k
+
+        traceWith tracer . RunLocalServer =<< Snocket.getLocalAddr diNtcSnocket localSocket
+        Server.with
+          Server.Arguments {
+              sockets               = localSocket :| [],
+              snocket               = diNtcSnocket,
+              tracer                = dtLocalServerTracer,
+              connectionLimits      = localConnectionLimits,
+              inboundGovernorArgs   =
+                IG.Arguments {
+                  tracer                = dtLocalInboundGovernorTracer,
+                  transitionTracer      = nullTracer,
+                  debugTracer           = nullTracer,
+                  connectionDataFlow    = ntcDataFlow,
+                  idleTimeout           = Nothing,
+                  withConnectionManager = localWithConnectionManager localInbInfoChannel,
+                  mkConnectionHandler   = mkLocalConnectionHandler,
+                  infoChannel           = localInbInfoChannel } }
```

Comment:
I find it more readable to put the final brackets in separate lines, so it's easier to see when a given record finishes (especially in nested records).
```suggestion
                  infoChannel           = localInbInfoChannel
                }
              }
```